infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -611,9 +598,38 @@ fun splitLocationTrack(
                 operation = target.getOperation(),
             )
         }
-        .also { result -> validateSplitResult(result, geometry) }
+        .also { result -> validateSplitResult(result, sourceGeometry) }
+}
+
+private fun connectPartialDuplicateEdges(
+    geometry: LocationTrackGeometry,
+    replacements: List<LayoutEdge>,
+    replacementIndices: IntRange,
+): List<LayoutEdge> {
+    require(replacements.isNotEmpty()) { "Cannot replace edges with nothing" }
+    // The connecting edges (edges around the connection point) must be adjusted to fit the new geometry properly
+    val startConnection =
+        geometry.edges.getOrNull(replacementIndices.first - 1)?.let { e ->
+            e.connectEndTo(replacements.first(), MAX_SPLIT_GEOM_ADJUSTMENT)
+                ?: failEdgeConnection(e, replacements.first())
```

Comment:
Tämä sisempi null (eli tuon connect..-funkkarin paluuarvo null) tarkoittaa että ei voitu yhdistää -> tästä heitto. Tuo voi käytännössä syntyä lähinnä jos tätä kutsuttaisi jollain todella epäyhteensopivilla edge replacementeilla.

Sen sijaan ulompi null (gemetry.edges.getOrNull) tarkoittaa että ei ole mitään edgeä ennen yhdyskohtaa -> nothing to do.